### ISSUE TITLE
[Feature Request] Add reload and sessions and authentication in Robyn

### ISSUE THREAD

**Issue by sansyrox** [sansyrox/robyn#164]
**Current Behavior**

Robyn is missing a lot of features in regards to authentication and authorisation. Implement them.

**Desired Behavior**

---
**Screenshots / Mockups**

**Alternatives**

---

**Comment by sansyrox** [comment]
Update: reload (--dev) and authentication (configure_authentication, documented in authentication.mdx) are both implemented, and cookie support (set_cookie) is available to build sessions on. The remaining item from this umbrella issue is first-class session management, tracked in #945 — leaving this open to track that, or it can be closed in favor of #945.

**Comment by sansyrox** [comment]
Closing this 2022 umbrella in favor of the more specific issues — every item is now done or tracked:

- **reload** — implemented (`--dev`).
- **authentication** — implemented (`configure_authentication` + `AuthenticationHandler`, documented in `authentication.mdx`).
- **cookies** — available (`Response.set_cookie`).
- **sessions** — the only remaining item, tracked in #945 and implemented in #1411 (`request.session` + `app.configure_sessions`), which closes #945.

Once #1411 lands there's nothing left here that isn't tracked in #945, so closing in favor of #945.
